Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/edit/ContactDiaryEditPersonsViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package de.rki.coronawarnapp.contactdiary.ui.edit
+
+import androidx.lifecycle.asLiveData
+import com.squareup.inject.assisted.AssistedInject
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryPerson
+import de.rki.coronawarnapp.contactdiary.storage.entity.ContactDiaryPersonEntity
+import de.rki.coronawarnapp.contactdiary.storage.entity.toContactDiaryPersonEntity
+import de.rki.coronawarnapp.contactdiary.storage.repo.ContactDiaryRepository
+import de.rki.coronawarnapp.ui.SingleLiveEvent
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import kotlinx.coroutines.flow.map
+
+class ContactDiaryEditPersonsViewModel @AssistedInject constructor(
+    private val contactDiaryRepository: ContactDiaryRepository,
+    dispatcherProvider: DispatcherProvider
+) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
+
+    val personsLiveData = contactDiaryRepository.people.asLiveData()
```

Comment:
I think the database access can happen on the main thread here.

Reply:
Same for the other repo access calls below in this file.

Reply:
Is there a benefit to having it on the main thread?

Reply:
I worded it incorrectly, sorry, I meant: We should make sure to not access the database on the main thread.

Reply:
Ok. I thought you were asking to have it on the main thread.

Reply:
https://github.com/corona-warn-app/cwa-app-android/pull/1954/files